impl/src/test/java/org/ehcache/impl/internal/executor/PooledExecutionServiceTest.java

```diff
@@ -105,4 +108,23 @@ public void testGetOrderedExecutorSucceedsOnExistingDefaultPool() throws Excepti
     pooledExecutionService.stop();
   }
 
+  @Test
+  public void testAllThreadsAreStopped() throws Exception {
+    PooledExecutionServiceConfiguration configuration = new PooledExecutionServiceConfiguration();
+    configuration.addDefaultPool("dflt", 0, 1);
+    PooledExecutionService pooledExecutionService = new PooledExecutionService(configuration);
+    pooledExecutionService.start(null);
+    pooledExecutionService.getScheduledExecutor("dflt")
+      .schedule(new Runnable() {
+        @Override
+        public void run() {
+
+        }}, 10, TimeUnit.HOURS);
+    pooledExecutionService.stop();
+
+    Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
+    for(Thread thread : threadSet) {
+      assertThat(thread.getName()).doesNotContain("pool");
```

Comment:
This only matches the string from the underlying ThreadPoolExecutor... we should probably also validate that the "dflt" pool threads have gone away.  (Which would also implying starting them with a task that gets run before we shutdown).